From 8738765da56bcbaab8d46a2fc910692d8cf712df Mon Sep 17 00:00:00 2001
From: "Liu, Kai1" <kai1.liu@intel.com>
Date: Wed, 19 Jun 2019 16:50:04 +0800
Subject: [PATCH 7/8] Modify to build out codec2 component libraries

This is a workaround patch to generate codec2 components libraries
on x86_64 platform.

Change-Id: If236e24e09da57c20fd4efa93621c53e1624002b
Tracked-On: https://jira.devtools.intel.com/browse/OAM-80593
Signed-off-by: Liu, Kai1 <kai1.liu@intel.com>
---
 media/codec2/components/base/Android.bp  | 3 ---
 media/codec2/sfplugin/utils/Android.bp   | 1 -
 media/libstagefright/flac/dec/Android.bp | 1 -
 3 files changed, 5 deletions(-)

diff --git a/media/codec2/components/base/Android.bp b/media/codec2/components/base/Android.bp
index f10835f366..2932af4f81 100644
--- a/media/codec2/components/base/Android.bp
+++ b/media/codec2/components/base/Android.bp
@@ -30,7 +30,6 @@ cc_library_shared {
             "unsigned-integer-overflow",
             "signed-integer-overflow",
         ],
-        cfi: true,
     },

     ldflags: ["-Wl,-Bsymbolic"],
@@ -78,7 +77,6 @@ cc_defaults {
             "unsigned-integer-overflow",
             "signed-integer-overflow",
         ],
-        cfi: true,
     },
 }

@@ -91,7 +89,6 @@ cc_defaults {
         misc_undefined: [
             "signed-integer-overflow",
         ],
-        cfi: true,
     },
 }

diff --git a/media/codec2/sfplugin/utils/Android.bp b/media/codec2/sfplugin/utils/Android.bp
index 8c8f025aba..5c5d2e9ac1 100644
--- a/media/codec2/sfplugin/utils/Android.bp
+++ b/media/codec2/sfplugin/utils/Android.bp
@@ -31,7 +31,6 @@ cc_library_shared {
     ],

     sanitize: {
-        cfi: true,
         misc_undefined: [
             "unsigned-integer-overflow",
             "signed-integer-overflow",
diff --git a/media/libstagefright/flac/dec/Android.bp b/media/libstagefright/flac/dec/Android.bp
index b494e16a99..3c86bc5d8e 100644
--- a/media/libstagefright/flac/dec/Android.bp
+++ b/media/libstagefright/flac/dec/Android.bp
@@ -15,7 +15,6 @@ cc_library {
             "signed-integer-overflow",
             "unsigned-integer-overflow",
         ],
-        cfi: true,
     },

     static: {
--
2.21.0
